src/Database/Connection.php

```diff
@@ -380,7 +380,7 @@ public function getSchemaCollection(): SchemaCollectionInterface
         if (!empty($this->_config['cacheMetadata'])) {
             return $this->_schemaCollection = new CachedCollection(
                 new SchemaCollection($this),
-                $this->configName(),
+                empty($this->_config['cacheKeyPrefix']) ? $this->configName() : $this->_config['cacheKeyPrefix'],
```

Comment:
I think [CachedCollection::cacheKey()](https://api.cakephp.org/3.8/source-class-Cake.Database.Schema.CachedCollection.html#72-81) that is used for metadata caching in [CachedCollection::describe()](https://api.cakephp.org/3.8/source-class-Cake.Database.Schema.CachedCollection.html#54) should be adjusted. 

Something like `return $this->_connection->getCacheKeyPrefix() . '_' . $name;` instead `return $this->_connection->configName() . '_' . $name;`.

Reply:
Let me give you an example. We have cache keys like `_cake_model_client_33_areas` where `_cake_model_` is cache prefix as configured in app.php, `client_33` is the connection name, `areas` is the table name. What is needed is to set cache metadata key to `client` so that the final cache key would be `_cake_model_client_areas`.

Reply:
That should affect only metadata cache keys. Cache keys to any tables other than in `information_schema` should not be affected. Queries to different client databases return different results.

Reply:
This change exactly what you want, it's just that it's targetted at 4.x not 3.x.